src/backend/distributed/operations/shard_cleaner.c

```diff
@@ -131,32 +131,62 @@ DropMarkedShards(bool waitForCleanupLock)
 			continue;
 		}
 
-		ShardPlacement *shardPlacement = LoadShardPlacement(placement->shardId,
-															placement->placementId);
-		ShardInterval *shardInterval = LoadShardInterval(shardPlacement->shardId);
-
-		ereport(LOG, (errmsg("dropping shard placement " INT64_FORMAT " of shard "
-							 INT64_FORMAT " on %s:%d after it was moved away",
-							 shardPlacement->placementId, shardPlacement->shardId,
-							 shardPlacement->nodeName, shardPlacement->nodePort)));
+		if (TryDropShard(placement))
+		{
+			removedShardCount++;
+		}
+	}
 
-		/* prepare sql query to execute to drop the shard */
-		StringInfo dropQuery = makeStringInfo();
-		char *qualifiedTableName = ConstructQualifiedShardName(shardInterval);
-		appendStringInfo(dropQuery, DROP_REGULAR_TABLE_COMMAND, qualifiedTableName);
+	return removedShardCount;
+}
 
-		List *dropCommandList = list_make2("SET LOCAL lock_timeout TO '1s'",
-										   dropQuery->data);
 
-		/* remove the shard from the node and the placement information */
+/*
+ * TryDropShard tries to drop the given shard placement and returns
+ * true on success. On failure, this method swallows errors and emits them
+ * as WARNINGs.
+ */
+static bool
+TryDropShard(GroupShardPlacement *placement)
+{
+	MemoryContext savedContext = CurrentMemoryContext;
+	ShardPlacement *shardPlacement = LoadShardPlacement(placement->shardId,
+														placement->placementId);
+	ShardInterval *shardInterval = LoadShardInterval(shardPlacement->shardId);
+
+	ereport(LOG, (errmsg("dropping shard placement " INT64_FORMAT " of shard "
+						 INT64_FORMAT " on %s:%d after it was moved away",
+						 shardPlacement->placementId, shardPlacement->shardId,
+						 shardPlacement->nodeName, shardPlacement->nodePort)));
+
+	/* prepare sql query to execute to drop the shard */
+	StringInfo dropQuery = makeStringInfo();
+	char *qualifiedTableName = ConstructQualifiedShardName(shardInterval);
+	appendStringInfo(dropQuery, DROP_REGULAR_TABLE_COMMAND, qualifiedTableName);
+
+	List *dropCommandList = list_make2("SET LOCAL lock_timeout TO '1ms'",
```

Comment:
This used to be 1s, 1ms seems too short / might trigger just because the process stalls a bit.

This line could also use a comment saying why we take the lock timeout.